File: app/src/main/java/kittentrate/game/GameContract.java
package kittentrate.game;

import android.widget.ViewFlipper;

import java.util.List;

import kittentrate.data.repository.model.PhotoEntity;
import kittentrate.score.PlayerScore;

/**
 * Created by Manuel Lorenzo on 21/03/2017.
 */

public interface GameContract {
    interface View {
        void showLoadingView();

        void hideLoadingView();

        void turnCardsOver();

        void notifyAdapterItemChanged(int pos);

        void notifyAdapterItemRemoved(String id);

        void setAdapterData(List<PhotoEntity> photoEntityList);

        void removeViewFlipper();

        void showErrorView();

        boolean shouldDispatchTouchEvent();

        void onScoreChanged(int gameScore);

        void checkGameFinished();
    }

    interface Presenter {
        void onItemClicked(int position, Card card, ViewFlipper viewFlipper);

        void removeCardsFromMaps();

        void start();

        void onScoredEntered(PlayerScore playerScore);

        void notifyAdapterItemChanged(Integer key);

        void onTurnCardsOver();

        void removeViewFlipper();

        void onGameScoreChanged(int gameScore);

        void notifyAdapterItemRemoved(String id);

        void onKittensMenuItemClicked();

        void onPuppiesMenuItemClicked();
    }
}

File: app/src/main/java/kittentrate/game/GameContract.kt
package kittentrate.game

import android.widget.ViewFlipper

import kittentrate.data.repository.model.PhotoEntity
import kittentrate.score.PlayerScore

/**
 * Created by Manuel Lorenzo on 21/03/2017.
 */

interface GameContract {
    interface View {
        fun showLoadingView()

        fun hideLoadingView()

        fun turnCardsOver()

        fun notifyAdapterItemChanged(pos: Int)

        fun notifyAdapterItemRemoved(id: String)

        fun setAdapterData(photoEntityList: List<PhotoEntity>)

        fun removeViewFlipper()

        fun showErrorView()

        fun shouldDispatchTouchEvent(): Boolean

        fun onScoreChanged(gameScore: Int)

        fun checkGameFinished()
    }

    interface Presenter {
        fun onItemClicked(position: Int, card: Card, viewFlipper: ViewFlipper)

        fun removeCardsFromMaps()

        fun start()

        fun onScoredEntered(playerScore: PlayerScore)

        fun notifyAdapterItemChanged(key: Int?)

        fun onTurnCardsOver()

        fun removeViewFlipper()

        fun onGameScoreChanged(gameScore: Int)

        fun notifyAdapterItemRemoved(id: String)

        fun onKittensMenuItemClicked()

        fun onPuppiesMenuItemClicked()
    }
}
